mlir/lib/Dialect/Tensor/Transforms/FoldTensorSubsetOps.cpp

```diff
@@ -154,6 +163,17 @@ LogicalResult InsertSliceOfTransferWriteOpFolder::matchAndRewrite(
   return success();
 }
 
+bool InsertSliceOfTransferWriteOpFolder::doesTransferWriteCoverInsertSlice(
+    vector::TransferWriteOp writeOp) {
+  if (writeOp.getShapedType().hasStaticShape())
+    return llvm::equal(writeOp.getVectorType().getShape(),
+                       writeOp.getShapedType().getShape());
```

Comment:
Shouldn't we move the check to `preconditionsFoldExtractOrInsertWithTransferOp`? Otherwise the preconditions and support could diverge quickly between read an write ops.

Reply:
Updated.

Reply:
I see. Yeah, makes sense. I did not think it carefully. I was also confused that why my comment was resolved but there are no related tests. Sorry for the bad suggestion. Can you revert it to what it was?